tests/table.spec.ts

```diff
@@ -139,6 +139,176 @@ test.describe('tests complex table operations', () => {
 
     // Check snapshot
     await expect(page.locator('.iris-grid-column')).toHaveScreenshot();
+
+    await page
+      .locator('.iris-grid')
+      .locator('data-testid=btn-page-close')
+      .first()
+      .click();
+
+    await expect(page.locator('.iris-grid .table-sidebar')).toHaveCount(0);
+  });
+
+  test('can conditional format', async ({ page }) => {
+    await page.goto('');
+    const consoleInput = page.locator('.console-input');
+    await consoleInput.click();
+
+    const command = `${makeTableCommand(
+      undefined,
+      TableTypes.StringAndNumber
+    )}`;
+
+    await pasteInMonaco(consoleInput, command);
+    await page.keyboard.press('Enter');
+
+    // Wait for the panel to show
+    await expect(page.locator('.iris-grid-panel')).toHaveCount(1);
+
+    // Wait until it's done loading
+    await expect(page.locator('.iris-grid-panel .loading-spinner')).toHaveCount(
+      0
+    );
+
+    // Model is loaded, need to make sure table data is also loaded
+    await expect(
+      page.locator('.iris-grid .iris-grid-loading-status')
+    ).toHaveCount(0);
+
+    const tableOperationsMenu = page.locator(
+      'data-testid=btn-iris-grid-settings-button-table'
+    );
+    await tableOperationsMenu.click();
+
+    // Wait for Table Options menu to show
+    await expect(page.locator('.table-sidebar')).toHaveCount(1);
+
+    // Open Conditional Formatting Panel
+    await page.locator('data-testid=menu-item-Conditional Formatting').click();
+
+    // Create New Formatting Rule
+    await page.getByRole('button', { name: 'Add New Rule' }).click();
+    await page.getByPlaceholder('Enter value').click();
+    await page.keyboard.type('3');
+    await page.getByRole('button', { name: 'No formatting' }).click();
+    await page.getByRole('button', { name: 'Positive' }).click();
+
+    const delayForGridRender = 500;
+    const conditionSelect = page.locator('#condition-select');
+
+    // Is Equal To
+    await page.getByRole('button', { name: 'Rows' }).click();
+    await page.waitForTimeout(delayForGridRender);
```

Comment:
We do not want to waitForTimeout, the [Playwright Docs explicitly discourage it's use](https://playwright.dev/docs/api/class-page#page-wait-for-timeout).

Part of the reason we don't want to is that it's flakey, e.g. it could take more than 500ms, and then this test would fail; or (what's more likely the case here) is that 500ms is way longer than it takes for the grid to re-render, so now our test is taking longer to run than necessary.

Instead, we should be asserting on things we can see on the page. There's a loading bar with text whenever we apply table operations (at least, there should be), so we can check the load is done by doing something like:
```
    // Wait until it's done loading
    await expect(
      page.locator('.iris-grid .iris-grid-loading-status')
    ).toHaveCount(0);
```
If that's not working, then we need to double-check the loading bar is appearing correctly in each case, or investigate what else is going on.

Reply:
Yea it seems that changing the conditional formatting rules doesn't trigger the `startLoading` function in IrisGrid.tsx and that's needed to show the loading status element. Adding in the implementation now but wanted to make sure this omission wasn't intended in the original PR #409